Source/SA/RFmx Instr/RFmxInstr.cs

```diff
@@ -0,0 +1,85 @@
+                return new InstrumentConfiguration
+                {
+                    LoConfigurations = new LocalOscillatorConfiguration[] { LocalOscillatorConfiguration.GetDefault() }       
+                };
+            }
+
+            public static InstrumentConfiguration GetDefault(RFmxInstrMX sessionHandle)
+            {
+                InstrumentConfiguration instrConfig = GetDefault(); // covers case for sub6 instruments with a single configurable LO
+                // lo configuration will now be overridden if the instrument has multiple configurable LOs
+                sessionHandle.GetInstrumentModel("", out string instrumentModel);
+                switch (instrumentModel)
+                {
+                    case "NI PXIe-5830":
+                    case "NI PXIe-5831":
+                        LocalOscillatorConfiguration lo1Config = LocalOscillatorConfiguration.GetDefault();
+                        lo1Config.ChannelName = "LO1";
+                        LocalOscillatorConfiguration lo2Config = LocalOscillatorConfiguration.GetDefault();
+                        lo2Config.ChannelName = "LO2";
+                        instrConfig.LoConfigurations = new LocalOscillatorConfiguration[] { lo1Config, lo2Config };
+                        break;
+                }
+                return instrConfig;
+            }
+        }
+        #endregion
+
+        #region Instrument Configurations
+        public static void ConfigureInstrument(RFmxInstrMX instrHandle, InstrumentConfiguration instrConfig)
+        {
+            foreach (LocalOscillatorConfiguration loConfig in instrConfig.LoConfigurations)
+            {
+                switch (loConfig.SharingMode)
+                {
+                    case LocalOscillatorSharingMode.None:
+                        instrHandle.SetAutomaticSGSASharedLO(loConfig.ChannelName, RFmxInstrMXAutomaticSGSASharedLO.Disabled);
+                        instrHandle.SetLOExportEnabled(loConfig.ChannelName, false);
+                        instrHandle.SetLOSource(loConfig.ChannelName, RFmxInstrMXConstants.LOSourceOnboard);
+                        break;
+                    case LocalOscillatorSharingMode.Manual:
+                        instrHandle.SetAutomaticSGSASharedLO(loConfig.ChannelName, RFmxInstrMXAutomaticSGSASharedLO.Disabled);
+                        instrHandle.SetLOExportEnabled(loConfig.ChannelName, loConfig.ExportEnabled);
+                        instrHandle.SetLOSource(loConfig.ChannelName, loConfig.Source);
+                        break;
+                    default: // default to automatic case
+                        instrHandle.SetAutomaticSGSASharedLO(loConfig.ChannelName, RFmxInstrMXAutomaticSGSASharedLO.Enabled);
+                        instrHandle.ResetAttribute(loConfig.ChannelName, RFmxInstrMXPropertyId.LOExportEnabled);
+                        instrHandle.ResetAttribute(loConfig.ChannelName, RFmxInstrMXPropertyId.LOSource);
+                        break;

```

Comment:
I was thinking about whether the automatic mode covers the different use cases that we have. I will create a general review note which covers that as well but here I just like to ask: Are we sure covering 5831 IF-RF? I know that the instrHandle.ConfigureAutomaticSGSASharedLO which works in connection with the playback library would handle the IF-RF case